prelude/slang-cpp-types.h

```diff
@@ -85,6 +85,10 @@ struct Matrix
     Vector<T, COLS> rows[ROWS];
 };
 
+// We can just map `NonUniformResourceIndex` type directly to the index type on CPU, as CPU does not require
+// any special handling around such accesses.
+typedef size_t NonUniformResourceIndex;
```

Comment:
Note that `NonUniformResourceIndex` is not a type in HLSL (or on the GPU targets), but rather a special-case identity function. It's kind of a semantic mess, with the main precedent being things like the `expect(...)` sorts of functions that are used for branch hinting (with the big difference being that those are about *hints* and thus do not affect correctness).